src/js/pages/Home.js

```diff
@@ -3,105 +3,103 @@ var $ = require('jquery');
+import Card from "../components/Card";
 
 import { NotificationStack } from 'react-notification';
-import { OrderedSet } from 'immutable';
 var moment = require('moment');
 
 export default class Featured extends React.Component {
   constructor(props) {
-    super();
-    this.addNotification = this.addNotification.bind(this);
-    this.removeNotification = this.removeNotification.bind(this);
+    super(props);
+    this.getEvents = this.getEvents.bind(this);
+    this.getNotifications = this.getNotifications.bind(this);
     this.state = {
-      messages: [],
-      messageStack: OrderedSet(),
+      notifications: [],
       events: [],
       apiBaseURL: props.route.apiBaseURL,
     };
-  }
+  };
 
-  addNotification (message) {
-    const currTime = moment().unix();
-    if (currTime > message.expirytime) {
-      return;
-    }
-    var self = this;
-    return this.setState({
-      messageStack: this.state.messageStack.add({
-        message: message.message,
-        key: message.id,
-        dismissAfter: (message.expirytime - currTime) * 1000
-      })
-    });
-  }
+  componentDidMount() {
+    NotificationStore.getAll();
+    EventStore.getAll();
+  };
 
-  removeNotification (count) {
-    this.setState({
-      messageStack: this.state.messageStack.filter(n => n.key !== count)
-    })
-  }
+  componentWillMount() {
+    NotificationStore.on("received", this.getNotifications);
+    NotificationStore.on("error", this.showNotificationStoreError);
+    EventStore.on("received", this.getEvents);
+    EventStore.on("error", this.showEventStoreError);
+  };
 
+  componentWillUnmount() {
+    NotificationStore.removeListener("received", this.getNotifications);
+    NotificationStore.removeListener("error", this.showNotificationStoreError);
+    EventStore.removeListener("received", this.getEvents);
+    EventStore.removeListener("error", this.showEventStoreError);
+  };
 
-  componentDidMount() {
-    var self = this;
-    this.broadcastRequest = $.get(this.state.apiBaseURL + "/broadcast", function (result) {
-      this.setState({
-        messages: result.data.map(function(message) {
-          return message.attributes;
-        })
-      });
+  removeNotification (id) {
+    this.setState({
+      notifications: this.state.notifications.filter(n => n.key !== id)
+    })
+  };
 
-      this.messageStack = new OrderedSet();
-      this.state.messages.forEach(function(m) {
-        self.addNotification(m);
+  getNotifications() {
+    var unexpired = NotificationStore.getUnexpired();
+    let temp = [];
+    const currTime = moment().unix();
+    unexpired.forEach(function(notification) {
+      temp.push({
+        message: notification.attributes.message,
+        key: "notification-" + notification.attributes.id,
+        action: 'Dismiss',
+        dismissAfter: (notification.attributes.expirytime - currTime) * 1000,
+        onClick: () => this.removeNotification("notification-" + notification.attributes.id),
```

Comment:
They do, the issue was that events may have the same id, so we build a type string to differentiate between them.